src/RedisCache.php

```diff
@@ -27,20 +30,64 @@
  */
 final class RedisCache implements CacheInterface
 {
+    /**
+     * @var array<ConnectionInterface>|ConnectionInterface $connections Predis connections instance to use
+     */
+    private ConnectionInterface|array $connections;
+
     /**
      * @param ClientInterface $client Predis client instance to use.
      */
     public function __construct(private ClientInterface $client)
     {
+        $this->connections = $this->client->getConnection();
     }
 
+    /**
+     * Checking Predis cluster usage
+     *
+     * @return bool
+     */
+    public function isCluster(): bool
+    {
+        /** @psalm-suppress MixedAssignment, PossibleRawObjectIteration */
+        foreach ($this->connections as $connection) {
+            /** @var StreamConnection $connection */
+            $cluster = (new Client($connection->getParameters()))->info('Cluster');
+            /** @psalm-suppress MixedArrayAccess */
+            if (isset($cluster['Cluster']['cluster_enabled']) && 1 === (int)$cluster['Cluster']['cluster_enabled']) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * @param string $key
+     * @param mixed|null $default
+     *
+     * @throws InvalidArgumentException
+     *
+     * @return mixed
+     */
     public function get(string $key, mixed $default = null): mixed
     {
         $this->validateKey($key);
+        /** @var string|null $value */
         $value = $this->client->get($key);
         return $value === null ? $default : unserialize($value);
     }
 
+    /**
+     * @param string $key
+     * @param mixed $value
+     * @param DateInterval|int|null $ttl
+     *
+     * @throws InvalidArgumentException
+     *
+     * @return bool
+     */
```

Comment:
```suggestion
    /**
     * @param string $key
     * @param mixed $value
     * @param DateInterval|int|null $ttl
     *
     * @throws InvalidArgumentException
     *
     * @return bool
     */
```

Should be inherited automatically from cache interface.